pkg/landscaper/installations/installation.go

```diff
@@ -48,14 +48,27 @@ func (i *Installation) ImportStatus() *ImportStatus {
 
 // GetImportDefinition return the import for a given key
 func (i *Installation) GetImportDefinition(key string) (lsv1alpha1.ImportDefinition, error) {
-	for _, def := range i.Blueprint.Info.Imports {
+	imports := getFlattenedImports(i.Blueprint.Info.Imports)
```

Comment:
@schrodit I'd like to have your opinion on this:
This check is called to see whether the parent satisfies all imports of the subinstallation. Before my changes, this looped over `i.Blueprint.Info.Imports` (over all direct imports of the parent) and checked whether the referenced import exists. With conditional imports, this is not enough anymore.

What I changed: It now loops over a flattened import list, which contains not only the direct import definitions of the parent, but also the conditional ones - independently of whether the condition is fulfilled or not.

The potential problem I see: This doesn't check whether the condition for conditional imports is fulfilled. So, in theory, an import definition could be found which doesn't actually exist, as it depends on another import which is not fulfilled. In practise, this shouldn't happen, as subinstallation imports that reference conditional parent imports are dynamically removed from the subinstallation template, if the condition which would provide the parent import is not fulfilled.

However, this still feels pretty 'dirty' and I'm not sure that I didn't miss a case in which this assumption breaks.
Should we build it in a way that the flattened list only contains conditional import definitions, if the corresponding condition is fulfilled? If so, how do we get the information which conditional imports are fulfilled at this point (`i.Imports` is `nil`)?
What do you think?